### ISSUE TITLE
Empty channel waiting forever when viewed and mixed

### ISSUE THREAD

**Issue by MaxUlysse** [nextflow-io/nextflow#428]
I managed to reproduce this strange bug on a simple example.
So I have two parameters, `foo` is the condition for the process to be executed, `bar` is the condition to display the output.

If this is run with no parameter, with `--foo` or with `--bar --foo`, it will run without problem.
But if this is run with only `--bar`, then one channel (I'm guessing `finalChannel`) is left hanging forever.

```groovy
foo = params.foo
bar = params.bar

process Process1 {
  publishDir "data", mode: 'copy'

  input:

  output:
    file("test.txt") into out

  when: foo

  script:
  """
  touch test.txt
  """
}

if (bar) out = out.view {"File: $it"}

finalChannel = Channel.empty().mix(out)
```

**Comment by pditommaso** [comment]
Squashed!

**Comment by pditommaso** [comment]
Currently available in `0.25.6-SNAPSHOT`
